airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/macros.py

```diff
@@ -126,5 +126,15 @@ def format_datetime(dt: Union[str, datetime.datetime], format: str) -> str:
     return dt_datetime.strftime(format)
 
 
-_macros_list = [now_utc, today_utc, timestamp, max, day_delta, duration, format_datetime]
+def format_datetime_string(dt: str, format: str) -> datetime.datetime:
+    """
+    Converts string to datetime based on input format
+
+    Usage:
+    `"{{ format_datetime_string(config.start_date, '%a, %d %b %Y %H:%M:%S %z') }}"`
+    """
+    return datetime.datetime.strptime(dt, format)
```

Comment:
Should we use our DatetimeParser here? It feels like it supports even more cases like `%s_as_float` and `%ms`

Reply:
we could, or even pendulum.from_format . LMK if this change is fundamental for this PR.